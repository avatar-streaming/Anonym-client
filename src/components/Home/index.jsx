import { Link } from "react-router-dom";
import useFetchStreamings from "../../hooks/useFetchStreamings";

function Home() {
  const streamings = useFetchStreamings();

  return (
    <div className="content-wrapper">
      <div className="streaming-list">
        {
          streamings.map((streaming) => (
            <Link
              to={`/streaming/${streaming.streamer._id}`}
              key={streaming._id}
              className="streaming"
            >
              <div className="streaming__top">
                <img
                  src={streaming.thumnail}
                  alt="streaming thumnail"
                  className="streaming__thumnail"
                  loading="lazy"
                />
              </div>
              <div className="streaming__bottom">
                <img
                  src={streaming.streamer.thumnail}
                  alt="streamer thumnail"
                  className="streaming__streamer-thumnail"
                />
                <div className="streaming__description">
                  <div className="streaming__title">{streaming.title}</div>
                  <div className="streaming__streamer-name">{streaming.streamer.userName}</div>
                </div>
              </div>
            </Link>
          ))
        }
      </div>
    </div>
  );
}

export default Home;
