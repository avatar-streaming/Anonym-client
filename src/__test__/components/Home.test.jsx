import React from "react";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import { Router } from "react-router";
import Home from "../../components/Home";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";
import { createMemoryHistory } from "history";

describe("<Home />", () => {
  const mockStore = configureMockStore([thunk]);
  const history = createMemoryHistory();
  let store = null;
  let component = null;

  beforeEach(() => {
    store = mockStore({
      streaming: {
        userStreaming: null,
        streamings: [],
        isLoading: false,
        err: null,
      },
    });

    component = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <Home />
        </Router>
      </Provider>
    );
  });

  it("should render with given state from redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
