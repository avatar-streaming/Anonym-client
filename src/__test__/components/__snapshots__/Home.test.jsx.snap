// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render with given state from redux store 1`] = `
<div
  className="content-wrapper"
>
  <div
    className="streaming-list"
  />
</div>
`;
